webserver/views/api/v1/core.py

```diff
@@ -126,33 +127,41 @@ def submit_low_level_nombid():
     except ValueError as e:
         raise webserver.views.api.exceptions.APIBadRequest("Cannot parse JSON document: %s" % e)
     
-    if 'musicbrainz_recordingid' in data['metadata']['tags'].keys():
-        mbid = data['metadata']['tags']['musicbrainz_recordingid']
-        action = "mbid_submitted"
-    elif 'musicbrainz_trackid' in data['metadata']['tags'].keys():
-        mbid = data['metadata']['tags']['musicbrainz_trackid']
-        action = "mbid_submitted"
+    if 'musicbrainz_recordingid' in data['metadata']['tags'].keys() or 'musicbrainz_trackid' in data['metadata']['tags'].keys():
+        raise webserver.views.api.exceptions.APIBadRequest('### The data files contains an ID this endopoint only accept nonmbid submissions!')
     elif 'md5_encoded' in data['metadata']['audio_properties'].keys():
```

Comment:
This is quite a complex method now. I would like to see a number of tests that confirm that various metadata values cause the correct lookups to be made. Every time you have a different control flow (for example, an exception or an if statement) you should consider having another test to verify the behaviour. Looking at your exceptions, here are some suggestions for tests:

* Bad json submitted to the endpoint
* A submission with a recording id tag
* A submission with no md5 value
* A submission with an md5 twice in a row
* A submission with no artist or title tag
* An error calling messybrainz
* A submission with additional metadata (album, track number, etc)

for the methods which call listenbrainz, you can create a `mock` of the messybrainz lookup method: http://www.voidspace.org.uk/python/mock/
so that you don't actually call this method during testing